pkg/ccl/changefeedccl/changefeed_dist_test.go

```diff
@@ -480,6 +482,64 @@ func (rdt *rangeDistributionTester) balancedDistributionUpperBound(numNodes int)
 	return int(math.Ceil((1 + rebalanceThreshold.Get(&rdt.lastNode.ClusterSettings().SV) + 0.1) * 64 / float64(numNodes)))
 }
 
+func TestChangefeedWithDistributionStrategyOptions(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
+	clusterOptions := []changefeedbase.ChangefeedRangeDistributionStrategy{
+		changefeedbase.ChangefeedRangeDistributionStrategyDefault,
+		changefeedbase.ChangefeedRangeDistributionStrategyBalancedSimple,
+	}
+	changefeedOptions := []changefeedbase.ChangefeedRangeDistributionStrategy{
+		changefeedbase.ChangefeedRangeDistributionStrategyDefault,
+		changefeedbase.ChangefeedRangeDistributionStrategyBalancedSimple,
+		changefeedbase.ChangefeedRangeDistributionStrategyNotSpecified,
+	}
+	testutils.RunValues(t, "cluster option", clusterOptions, func(t *testing.T, clusterOption changefeedbase.ChangefeedRangeDistributionStrategy) {
+		testutils.RunValues(t, "changefeed option", changefeedOptions, func(t *testing.T, changefeedOption changefeedbase.ChangefeedRangeDistributionStrategy) {
+			expectedStrat := changefeedOption
+			if changefeedOption == changefeedbase.ChangefeedRangeDistributionStrategyNotSpecified {
+				expectedStrat = clusterOption
+			}
+
+			testFn := func(t *testing.T, s TestServer, f cdctest.TestFeedFactory) {
+				sqlDB := sqlutils.MakeSQLRunner(s.DB)
+				sqlDB.Exec(t, fmt.Sprintf("SET CLUSTER SETTING changefeed.default_range_distribution_strategy = '%s'", clusterOption))
+				sqlDB.Exec(t, `CREATE TABLE foo (a INT PRIMARY KEY, b STRING)`)
+				sqlDB.Exec(t, `INSERT INTO foo VALUES (0, 'initial')`)
+
+				stratCh := make(chan changefeedbase.ChangefeedRangeDistributionStrategy, 1)
+
+				knobs := s.TestingKnobs.DistSQL.(*execinfra.TestingKnobs).Changefeed.(*TestingKnobs)
+				knobs.RangeDistributionStrategyCallback = func(strat changefeedbase.ChangefeedRangeDistributionStrategy) {
+					select {
+					case stratCh <- strat:
+					default:
+						fmt.Printf("skipping strat: %s\n", strat)
+					}
+				}
+
+				createStmt := "CREATE CHANGEFEED FOR foo WITH initial_scan='no'"
+				if changefeedOption != changefeedbase.ChangefeedRangeDistributionStrategyNotSpecified {
+					createStmt += fmt.Sprintf(", range_distribution_strategy='%s'", changefeedOption)
+				}
+				feed := feed(t, f, createStmt)
+				defer closeFeed(t, feed)
+				testutils.SucceedsSoon(t, func() error {
+					select {
+					case strat := <-stratCh:
+						require.Equal(t, expectedStrat, strat)
+						return nil
+					default:
+						return errors.New("no range distribution strategy callback found")
+					}
+				})
+			}
+			cdcTest(t, testFn, feedTestForceSink("kafka"))
```

Comment:
let's not restrict the sink type so we can get more coverage

```suggestion
			cdcTest(t, testFn)
```